src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineDraft.java

```diff
@@ -0,0 +1,240 @@
+  }
+
+  @Then("Check url for draftId string")
+  public void checkUrlHasDraftId() {
+    Helper.waitSeconds();
+    boolean hasDraftId = Helper.urlHasString("draftId");
+    Assert.assertTrue(hasDraftId);
+  }
+
+  @Then("Reload the page")
+  public void reloadPage() {
+    Helper.reloadPage();
+  }
+
+  @Then("Export the pipeline")
+  public void exportPipeline() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("pipeline-export-btn"));
+    // Wait for the download to finish
+    Helper.waitSeconds(3);
+  }
+
+  @Then("Verify the exported pipeline")
+  public void verifyExportedJsonPipeline() throws IOException {
+    Reader reader = Files.newBufferedReader(Paths.get(
+      Constants.DOWNLOADS_DIR + pipelineName + "-" + PIPELINE_TYPE + ".json"));
+    JsonObject jsonObject = GSON.fromJson(reader, JsonObject.class);
+    JsonArray stages = jsonObject.getAsJsonObject("config").getAsJsonArray("stages");
+    String datasetProject = "";
+    String datasetName = "";
+    for (JsonElement node : stages) {
+      JsonObject nodeObj = node.getAsJsonObject();
+      JsonObject nodeProps = nodeObj.get("plugin").getAsJsonObject();
+      String nodeType = nodeProps.get("type").getAsString();
+      String nodeName = nodeProps.get("name").getAsString();
+      if (nodeName.equals(Commands.simpleSourceNode.getNodeName()) &&
+        nodeType.equals(Commands.simpleSourceNode.getNodeType())) {
+        JsonObject pluginProps = nodeProps.get("properties").getAsJsonObject();
+        datasetProject = pluginProps.get("datasetProject").getAsString();
+        datasetName = pluginProps.get("dataset").getAsString();
+      }
+    }
+    Assert.assertEquals(datasetName, DUMMY_DATASET);
+    Assert.assertEquals(datasetProject, DUMMY_PROJECT);
+    reader.close();
+  }
+
+  @Then("import pipeline from json {string}")
+  public void importPipeline(String jsonFile) {
+    Helper.importPipelineFromJson(jsonFile, pipelineDraftName);
+  }
+
+  @Then("Navigate to pipeline drafts page and sort by latest")
+  public void navigateToDraftsPage() throws InterruptedException {
+    SeleniumDriver.openPage(Constants.BASE_STUDIO_URL + "pipelines/drafts");
+    WaitHelper.waitForPageToLoad();
+    WebElement lastSavedColumnHeader = Helper.locateElementByXPath(
+      "//div[@data-testid='draft-pipeline-table']//*[contains(@class, 'grid-header')]//*[contains(text(),'Last saved')]"
+    );
+    // Click twice to sort by latest
+    ElementHelper.clickOnElement(lastSavedColumnHeader);
+    ElementHelper.clickOnElement(lastSavedColumnHeader);
+  }
+
+  @Then("Deploy the last saved pipeline")
+  public void deployTheNewestDraftPipeline() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId(("draft-" + pipelineDraftName)));
+
+    ElementHelper.clickOnElementUsingJsExecutor(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId("deploy-pipeline-btn")));
+    Helper.waitSeconds(10);
```

Comment:
I don't think we need to have explicit wait for deployment. You can just have some assert for pipeline status = deployed exists